### ISSUE TITLE
Chat: Best-Of N effect

### ISSUE THREAD

**Issue by enricoros** [enricoros/big-AGI#381]
Add a Chat Effect to the Chat Application, where the user can invoke an assistant completion on a "Best-of" mode.

In this mode, the assistant will generate N (2..10) completions, which will be shown to the user on a modal popup dialog. When all of them are done, they become active and scrollable. At this point either the user can select one of them to append to the conversation that started it all, or can use a "Judge" function specifying the criteria.

Requirements:

- Initiation
  - [x] initiate from the "Retry" or "Restart from here" menu options
  - [x] also initiate as a special "Mode" for the Chat
- [ ] at the ChatApp (or Message List?) termination, activate a "mode" that will display the chat and send it the instructions as initialState (which won't change anytime, and if it doesn't it's a new operation)
- [ ] **Alternatives** are generated from: Chat up to that point, submitted N times in parallel
- [ ] **Judging** happens by applying: [judge prompt, pre-judging instruction* with the "-1"-message inside, options(1..N), post-judging instruction]
  - [ ] *: are the pre/post judging instructions generated as well? 
- Rendering
  - [ ] Alternatives: 'warning' to show the in-progress status, and neutral/success at the end
  - [ ] Alternatives: inside render using <ChatMessage> over the in-progress text, to benefit from auto-everything
  - [ ] Judge: make it clear that the user needs to specify the "success criteria/what to look for"
  - [ ] While judging, the display of the N collapses, and one unified display comes out from it
- [ ] At the end add the message to the originating source, possibly via a onSuccess/onFail/onCancel callback strategy

**Comment by enricoros** [comment]
Duplicate of #470
